test/box/tx_man_intervals.test.lua

```diff
@@ -0,0 +1,146 @@
+env = require('test_run')
+test_run = env.new()
+test_run:cmd("create server tx_man with script='box/tx_man.lua'")
+test_run:cmd("start server tx_man")
+test_run:cmd("switch tx_man")
+
+txn_proxy = require('txn_proxy')
+
+tx1 = txn_proxy.new()
+tx2 = txn_proxy.new()
+tx3 = txn_proxy.new()
+
+s = box.schema.space.create('test')
+
+i = s:create_index('pk', {parts={{1, 'uint'}}})
+
+-- Completely empty selects
+tx1:begin()
+tx2:begin()
+tx1('s:select{}')
+tx2('s:select{}')
+tx1('s:replace{1, 1}')
+tx2('s:replace{2, 2}')
+tx1:commit()
+tx2:commit()
+
+s:truncate()
+
+-- Non-empty selects
+s:replace{3, 3}
+
+tx1:begin()
+tx2:begin()
+tx1('s:select{}')
+tx2('s:select{}')
+tx1('s:replace{1, 1}')
+tx2('s:replace{2, 2}')
+tx1:commit()
+tx2:commit()
+
+s:truncate()
+
+-- Insert then delete
+s:replace{3, 3}
+s:delete{3}
+
+tx1:begin()
+tx2:begin()
+tx1('s:select{}')
+tx2('s:select{}')
+tx1('s:replace{1, 1}')
+tx2('s:replace{2, 2}')
+tx1:commit()
+tx2:commit()
```

Comment:
I try to do this test case for 'hash' index type and got other results:
os.execute("rm -rf *.snap *.xlog")

box.cfg{memtx_use_mvcc_engine  = true}

txn_proxy = require('txn_proxy')

tx1 = txn_proxy.new()
tx2 = txn_proxy.new()

s = box.schema.space.create('test')
s:format({
         {name = 'id', type = 'unsigned'},
         })

i = s:create_index('pk', {type = 'tree', parts = {'id'}})

s:replace{3, 3}

tx1:begin()
tx2:begin()
tx1('s:select{}')
tx2('s:select{}')
tx1('s:replace{1, 1}')
tx2('s:replace{2, 2}')
tx1:commit()
tx2:commit()

For type == tree tx2 fails (it's should be so), but for type == hash not!
As i see now this patch fix problem only for `tree` index type!